pkg/resourceoverrides/cloudids_overrides.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourceoverrides
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func GetCloudIDSEndpointResourceOverrides() ResourceOverrides {
+	return ResourceOverrides{
+		Kind: "CloudIDSEndpoint",
+		Overrides: []ResourceOverride{
+			mapEndpointIpToEndpointIP(),
+		},
+	}
+}
+
+func mapEndpointIpToEndpointIP() ResourceOverride {
+	return ResourceOverride{
+		CRDDecorate: func(crd *apiextensions.CustomResourceDefinition) error {
+			for _, version := range crd.Spec.Versions {
+				schema := version.Schema.OpenAPIV3Schema
+				status, ok := schema.Properties["status"]
+				if !ok {
+					return fmt.Errorf("status field not found for version %s", version.Name)
+				}
+				observedState, ok := status.Properties["observedState"]
+				if !ok {
+					return fmt.Errorf("observedState field not found for version %s", version.Name)
+				}
+				endpointIp, ok := observedState.Properties["endpointIp"]
+				if !ok {
+					return fmt.Errorf("endpointIp field not found for version %s", version.Name)
+				}
+
+				// Rename endpointIp to endpointIP
+				observedState.Properties["endpointIP"] = endpointIp
+				delete(observedState.Properties, "endpointIp")
+			}
+
+			return nil
+		},
+		PostActuationTransform: func(original, reconciled *k8s.Resource, tfState *terraform.InstanceState, dclState *unstructured.Unstructured) error {
+			endpointIp, ok := reconciled.Status["endpointIp"]
+			if !ok {
+				return fmt.Errorf("endpointIp field was not populated")
```

Comment:
Why is it an error? I don't think we require all the output-only field be populated? Or maybe I miss anything?

Reply:
that's a good point! I thought we required all output only fields to be populated but it sounds like that's not the case today.

Reply:
AFAIK, output-only fields can be optional in general. We will need to thoroughly understand the API to make a decision.

Reply:
`endpointIP` is always present in this case I think. From the integ test `endpointIP` seems to be well populated in the managedFields too.